elastic/elastic-agent
# Make `k8sattributesprocessor` available in Agent running in OTel mode

## What does this PR do?


This PR makes the [`k8sattributesprocessor`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.102.0/processor/k8sattributesprocessor) available to users running Elastic Agent in `otel` mode.

## Why is it important?



To allow users of Elastic Agent running as as OpenTelemetry Collector to configure pipelines containing the `k8sattributesprocessor`.

## Checklist




## How to test this PR locally


1. Prepare the binary`DEV=true EXTERNAL=true SNAPSHOT=true PLATFORMS=linux/amd64 PACKAGES=docker mage -v package`
2. build the image `cd build/package/elastic-agent/elastic-agent-linux-amd64.docker/docker-build && docker build -t agent-otel-dev .`
3. Load to local cluster `kind load docker-image agent-otel-dev:latest`
4. prepare the ES secrets

```console
kubectl create secret generic elastic-secret \
     --from-literal=es_endpoint='https://test-otel-blablabla.es.us-central1.gcp.cloud.es.io/' \                 
     --from-literal=es_password='asdfgp' \     
     --from-literal=es_user='elastic'
```
5. specify the image:
 ```yaml
image: agent-otel-dev:latest
imagePullPolicy: Never
```
6. Deploy the manifest from https://github.com/elastic/opentelemetry-dev/pull/258 : ` k apply -f elastic-otel-collector.yaml`
7. Verify the output: 

```console
2024-06-10T13:17:25.677Z	info	ResourceLog #0
Resource SchemaURL: 
Resource attributes:
     -> k8s.namespace.name: Str(kube-system)
     -> k8s.pod.name: Str(kindnet-6lgrl)
     -> k8s.container.restart_count: Str(0)
     -> k8s.pod.uid: Str(3844b30e-67ee-4bcc-98ac-8f2196852939)
     -> k8s.container.name: Str(kindnet-cni)
     -> k8s.daemonset.name: Str(kindnet)
     -> k8s.node.name: Str(kind-control-plane)
     -> k8s.pod.start_time: Str(2024-06-10T13:09:03Z)
ScopeLogs #0
ScopeLogs SchemaURL: 
InstrumentationScope  
LogRecord #0
ObservedTimestamp: 2024-06-10 13:17:25.420729149 +0000 UTC
Timestamp: 2024-06-10 13:17:25.300601954 +0000 UTC
SeverityText: 
SeverityNumber: Unspecified(0)
Body: Str(I0610 13:17:25.300466       1 main.go:223] Handling node with IPs: map[172.21.0.2:{}])
Attributes:
     -> logtag: Str(F)
     -> log.file.path: Str(/var/log/pods/kube-system_kindnet-6lgrl_3844b30e-67ee-4bcc-98ac-8f2196852939/kindnet-cni/0.log)
     -> log.iostream: Str(stderr)
     -> time: Str(2024-06-10T13:17:25.300601954Z)
Trace ID: 
Span ID: 
Flags: 0
LogRecord #1
ObservedTimestamp: 2024-06-10 13:17:25.420987445 +0000 UTC
Timestamp: 2024-06-10 13:17:25.300633177 +0000 UTC
SeverityText: 
SeverityNumber: Unspecified(0)
Body: Str(I0610 13:17:25.300494       1 main.go:227] handling current node)
Attributes:
     -> log.iostream: Str(stderr)
     -> logtag: Str(F)
     -> log.file.path: Str(/var/log/pods/kube-system_kindnet-6lgrl_3844b30e-67ee-4bcc-98ac-8f2196852939/kindnet-cni/0.log)
     -> time: Str(2024-06-10T13:17:25.300633177Z)
Trace ID: 
Span ID: 
Flags: 0
	{"kind": "exporter", "data_type": "logs", "name": "debug"}
```
